neuralpp/symbolic/sympy_expression.py

```diff
@@ -81,6 +81,30 @@ def __init__(
         self._sympy_object = sympy_object
         self._type_dict = type_dict
 
+    def replace(self, from_expression: Expression, to_expression: Expression) -> Expression:
```

Comment:
moved `replace()` from `SymPyFunctionApplicationInterface` to here because it should be in `SymPyExpression` if it's about sympy specific replace().